lib/facter/framework/core/fact/internal/internal_fact_manager.rb

```diff
@@ -47,5 +56,59 @@ def resolve(searched_facts)
       resolved_facts.flatten!
       FactAugmenter.augment_resolved_facts(searched_facts, resolved_facts)
     end
+
+    def start_threads(searched_facts)
+      threads = []
+      # only resolve a fact once, even if multiple search facts depend on that fact
+      searched_facts
+        .uniq { |searched_fact| searched_fact.fact_class.name }
+        .each do |searched_fact|
+        threads << Thread.new do
+          resolve_fact(searched_fact)
+        end
+      end
+
+      threads
+    end
+
+    def join_threads(threads, searched_facts)
+      resolved_facts = []
+
+      threads.each do |thread|
+        thread.join
+        resolved_facts << thread.value unless thread.value.nil?
+      end
+
+      resolved_facts.flatten!
+
+      FactAugmenter.augment_resolved_facts(searched_facts, resolved_facts)
+    end
+
+    def thread_pool(searched_facts)
+      require 'concurrent'
+
+      pool = Concurrent::FixedThreadPool.new(12)
```

Comment:
number of threads could be configurable

Reply:
Removed the thread pool as it was the slowest implementation.

Reply:
I think this depends on the system that it runs on. Not sure that removing the thread poll entirely is a good idea.

Reply:
The code with the thread pool was never used and could not be enabled. If we want to try the thread pool in the future, we can revert commit: 3115202ecdc26cffed1abe3b30e1f596081e4a27